### PR TITLE
Documented/Indented whole codebase!

### LINKED ISSUE #378: GCI Task: Code Formatting
- **Objective**: Many python scripts don't have proper indentation and comments.The task is to choose one file and leave comments in the code after the proper understanding of the code and remove empty and unnecessary lines.The overall aim is that the code looks neat and tidy.

- **Requirements**: Code formatting, Python, Git
- **Expected Outcome**: A pull request with a properly formatted and neat version of the code of any one file of the codebase.
- **Type**: Code
- **Time to Complete the task in days** : 5

### PR DESCRIPTION
**Pull request by 0x48piraj**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- **Documented all the Python Scripts** :+1: 
- **Fixed many typo(s)** by GCI students _(eg. variable -> vatiable, file-> fils, ...)_ 
- Tested, read Python Documentation to ensure each comment is to the mark. :100: 
- proofread'ed each and every existing comment :smile:
- re-ordered/re-phrased many comments. :1234: 

---

**Screenshots :** _(Some Small references/proof(s)) :_

image

image

image

### CODE REVIEW

File: eventyay_export_anonymizer.py [hunk 1]
```diff
@@ -10,44 +10,65 @@
...
 
 with open(file, encoding="UTF-8") as f:
+#open up the file
     export = json.load(f)
+	#load the json to a variable "export"
 
 
 def walk(e):
     if isinstance(e, list):
+	# checks if the object (first argument) is an instance or subclass of classinfo class (second argument)
```
[the remark this answers is not in the document]
Comment (0x48piraj) [reply eventyay_export_anonymizer.py]: Fixed ! :+1:

File: eventyay_export_anonymizer.py [hunk 2]
```diff
@@ -10,44 +10,65 @@
...
+		# append the random variable `c` to `result`
     return "".join(result)
+	# return the final output
 
 
 walk(export)
+# use the function walk()
 
 with open(anonymous_file, 'w', encoding="UTF-8") as f:
+# open `anonymous_file`
```
[the remark this answers is not in the document]
Comment (0x48piraj) [reply eventyay_export_anonymizer.py]: Fixed ! :+1:

File: app/merge_badges_v2.py [hunk 3]
```diff
@@ -61,10 +68,14 @@ def generate_pdfs(folder_path):
...
 final_path = os.path.join(BADGES_FOLDER, 'all-badges.pdf')
+# Get final badge path
 pdfs = [file for file in os.listdir(BADGES_FOLDER) if file.lower().endswith('.pdf')]
+# check for .pdf files in BADGES_FOLDER
 merger = PdfFileMerger()
 for pdf in pdfs:
     merger.append(open(os.path.join(BADGES_FOLDER, pdf), 'rb'))
 
 with open(final_path, 'wb') as fout:
+# open final path file
```
[the remark this answers is not in the document]
Comment (0x48piraj) [reply app/merge_badges_v2.py]: Done :+1:

File: app/generate-badges.py [hunk 4]
```diff
@@ -10,17 +10,24 @@
...
 paper_sizes = {}
+# Initialize a list
 paper_sizes['A3'] = ['297mm', '420mm']
 paper_sizes['A4'] = ['210mm', '297mm']
 
 input_files = [file for file in os.listdir(UPLOAD_FOLDER) if file.lower().endswith(".csv")]
+# check for .csv files in UPLOAD_FOLDER
 
 with open(APP_ROOT + "/../badges/8BadgesOnA3.svg", encoding="UTF-8") as f:
+# open the file "8BadgesOnA3.svg"
```
[the remark this answers is not in the document]
Comment (0x48piraj) [reply app/generate-badges.py]: Done :+1:

File: app/generate-badges.py [hunk 5]
```diff
@@ -93,15 +100,18 @@ def generate_badges(aggregate, folder, index, picture, paper_size):
...
         os.makedirs(folder, exist_ok=True)
     except Exception:
         traceback.print_exc()
     ext = os.path.splitext(picpath)[1]
     badges_background = "badges_background{}".format(ext)
     shutil.copyfile(picpath, os.path.join(folder, badges_background))
+    # use shutil library to copy from `picpath` to `badges_background`
 
     with open(os.path.join(UPLOAD_FOLDER, input_file), encoding="UTF-8") as f:
+    # open the input file
```
[the remark this answers is not in the document]
Comment (0x48piraj) [reply app/generate-badges.py]: Done :+1:

**Comment by ms10398** [comment]
Please correct all Hound Errors

**Comment by 0x48piraj** [comment]
All done @ms10398 :+1: 
After each commit, if **hound** finds any error, posts it.
After my last commit, everything's fine. :100:

File: app/generate-badges.py [hunk 6]
```diff
@@ -10,17 +10,24 @@
 from lxml import etree
 
 NUMBER_OF_BADGES_PER_PAGE = 8
+# Initializing and setting global variable `NUMBER_OF_BADGES_PER_PAGE`
```
Comment (vaibhavsingh97) [inline app/generate-badges.py]: Hey! @0x48piraj we don't need comments for these small declarations

File: app/generate-badges.py [hunk 7]
```diff
@@ -10,17 +10,24 @@
...
 APP_ROOT = os.path.dirname(os.path.abspath(__file__))
+# Get root folder
```
Comment (vaibhavsingh97) [inline app/generate-badges.py]: Same

File: app/generate-badges.py [hunk 8]
```diff
@@ -10,17 +10,24 @@
...
 UPLOAD_FOLDER = os.path.join(APP_ROOT, 'static/uploads')
+# Get Upload Folder
 
 paper_sizes = {}
+# Initialize a list
```
Comment (vaibhavsingh97) [inline app/generate-badges.py]: same

File: app/generate_csv_eventyay.py [hunk 9]
```diff
@@ -7,11 +7,13 @@
 import os
 import sys
 
-# Get root folder
+
```
Comment (vaibhavsingh97) [inline app/generate_csv_eventyay.py]: Why you deleted it and changed position?

**Comment by 0x48piraj** [comment]
@vaibhavsingh97 
Ask @ParthS007 or look to the previous codebase. These changes are not mine, I have adapted them.
They were on `main.py`.
And the task is this only. To add documentation and indentation.
Refer : #378

**Comment by 0x48piraj** [comment]
It's updated now @open-event-bot

File: app/generate-badges.py [hunk 10]
```diff
@@ -10,18 +10,24 @@
 from lxml import etree
 
 NUMBER_OF_BADGES_PER_PAGE = 8
+# Initializing and setting global variable `NUMBER_OF_BADGES_PER_PAGE`
 APP_ROOT = os.path.dirname(os.path.abspath(__file__))
+# Get root folder
 UPLOAD_FOLDER = os.path.join(APP_ROOT, 'static/uploads')
+# Get Upload Folder
```
Comment (yashLadha) [inline app/generate-badges.py]: It is self explanatory, seeing it in constant notation. Try to not use words like `Get`.

File: app/generate-badges.py [hunk 11]
```diff
@@ -10,18 +10,24 @@
...
 paper_sizes = {}
+# Initialize a list
 paper_sizes['A3'] = ['297mm', '420mm']
 paper_sizes['A4'] = ['210mm', '297mm']
 
 input_files = [file for file in os.listdir(
     UPLOAD_FOLDER) if file.lower().endswith(".csv")]
 
 with open(APP_ROOT + "/../badges/8BadgesOnA3.svg", encoding="UTF-8") as f:
+    # open the file "8BadgesOnA3.svg"
```
Comment (yashLadha) [inline app/generate-badges.py]: Same as previous, these are self explanatory.

**Comment by ParthS007** [comment]
@0x48piraj  Please do suggestions as suggested by @vaibhavsingh97 and @yashLadha  and PR is ready to merge. 👍

**Comment by raju249** [comment]
I think the focus should be on..instead of adding single line comments which can be self explanatory, try adding multi-line comment at the top of the file and explain in brief what the file does, try explaining the functions in brief ( not too short, not too big)

Let me know what you guys think @0x48piraj @vaibhavsingh97 @ParthS007

**Comment by niccokunzmann** [comment]
A program usually does not include the purpose. This is something that should be commented.
A purpose is usually assigned to a function or other structural parts like classes and modules.
Please see [PEP257](https://www.python.org/dev/peps/pep-0257/) and [stackoverflow](https://stackoverflow.com/questions/3898572/what-is-the-standard-python-docstring-format) for how to use doc strings. This might be interesting: http://readthedocs.io/
Python usually uses Sphinx and Inter-sphinx to document projects. This way, you can reference Python modules, classes, variables, ... in other projects. e.g. `str` or `list`.

```
 APP_ROOT = os.path.dirname(os.path.abspath(__file__))
+# Get root folder
```
Here I would like to see something that helps understand why this variable should not be removed or which purpose it has:

```
# APP_ROOT is the folder of the badgeyay app. Use this to load files relative to the module
APP_ROOT = os.path.dirname(os.path.abspath(__file__))
```

**Comment by 0x48piraj** [comment]
Thanks a lot all you guys, and @niccokunzmann 

Now, I got a clear view.
Will do as suggested !

**Comment by harshit98** [comment]
@0x48piraj What's the status of this PR?

- Resolve conflicts.
- Squash your commits.
- Get it reviewed by rest of the mentors!
- You're PR will be merged! 🎉

**Comment by 0x48piraj** [comment]
WI'll do it in the auspicious day, 1st Jan :100:

**Comment by 0x48piraj** [comment]
Ping @vaibhavsingh97 @niccokunzmann @raju249 @harshit98 :tada: 

I took all measures, Please see https://github.com/fossasia/badgeyay/pull/449

**Comment by harshit98** [comment]
@0x48piraj Resolve conflicts.

**Comment by 0x48piraj** [comment]
Closing this!
I'll add document the code bit-by-bit @harshit98 :+1:

**Pull request closed without merging**